infra/build/functions/build_project.py

```diff
@@ -354,17 +354,22 @@ def get_build_steps_for_project(project,
     return []
 
   timestamp = get_datetime_now().strftime('%Y%m%d%H%M')
+
+  # if we use caching, then we need to use the right name. We assume that
+  # there is only a single sanitizer.
   if use_caching:
-    # Use cached built image.
-    build_steps = []
+    cache_name = ('us-central1-docker.pkg.dev/oss-fuzz/oss-fuzz-gen/'
```

Comment:
can we reuse the existing stuff here instead? https://github.com/google/oss-fuzz/blob/c30bd8ce00440dc4751f725e90a7a2817ee9716b/infra/build/functions/build_project.py#L189

We probably need to just remove lines 188-189, and just have:

```
  def cached_image(self):
    return _CACHED_IMAGE.format(name=self.real_name, sanitizer=self.cached_sanitizer)
```

We'll also have to delete lines 380-381 in this file, and just move the `project.cached_sanitizer = project.sanitizers[0]` here. Please also add a TODO here to support multiple sanitizers.

Reply:
> Please also add a TODO here to support multiple sanitizers.

I don't think we need this because we set it one at the time:

https://github.com/google/oss-fuzz/blob/edfa6730fa9f1dca77dd27c0b3a643171e082950/infra/build/functions/target_experiment.py#L61

and

https://github.com/google/oss-fuzz/blob/edfa6730fa9f1dca77dd27c0b3a643171e082950/infra/build/functions/target_experiment.py#L217